include/aws/mqtt/client.h

```diff
@@ -406,6 +406,21 @@ uint16_t aws_mqtt_client_connection_subscribe(
     aws_mqtt_suback_fn *on_suback,
     void *on_suback_ud);
 
+/**
+ * Resubscribe to all topics currently subscribed to. This is to help when resuming a connection with a clean session.
+ *
+ * \param[in] connection    The connection to subscribe on
+ * \param[in] on_suback     Called when a SUBACK has been received from the server and the subscription is complete
+ * \param[in] on_suback_ud  Passed to on_suback
+ *
+ * \returns The packet id of the subscribe packet if successfully sent, otherwise 0 (and aws_last_error() will be set).
+ */
+AWS_MQTT_API
+uint16_t aws_mqtt_resubscribe_existing_topics(
```

Comment:
Mention this in the comments/docs of `aws_mqtt_client_connection_set_connection_interruption_handlers` and aws_mqtt_client_on_connection_resumed_fn`. Something like

> If session_present is false, you probably want to call `aws_mqtt_resubscribe_existing_topics`.